### PR TITLE
Move userCardTemplateGateway methods in opfab UI API

### PR DESCRIPTION
**Pull request by freddidierRTE**
Nothing in release note as  #4438 is already in release note

### CODE REVIEW

File: src/docs/asciidoc/reference_doc/ui_api.adoc [hunk 1]
```diff
@@ -247,10 +248,280 @@ It can be used by a template to launch some processing when loading is complete
...
+
+The template can know the process id of the card by calling the _opfab.currentUserCard.getProcessId()_  function. The function will return a string corresponding to the process id.
+
 ```
-        opfab.currentCard.registerFunctionToGetUserResponse
+        const id = opfab.currentUserCard.getProcessId();
 ```
+=== Function getSelectedEntityRecipients
 
+The template can know the list of entities selected by the users as recipients of the card by calling the _opfab.currentUserCard.getSelectedEntityRecipients()_ function.  The function will return an array of entity ids.
```
Comment (ClementBouvierN) [inline src/docs/asciidoc/reference_doc/ui_api.adoc:328]: "selected by the users" -> "selected by the user"       singular 
"function.  The function" -> "function. The function"   one too many spaces

File: src/docs/asciidoc/reference_doc/ui_api.adoc [hunk 2]
```diff
@@ -247,10 +248,280 @@ It can be used by a template to launch some processing when loading is complete
...
+
+
+```
+        const recipients = opfab.currentUserCard.getSelectedEntityRecipients();
+```
+
+=== Function getSelectedEntityForInformationRecipients
+
+The template can know the list of entities selected by the users as recipients of the card by calling the _opfab.currentUserCard.getSelectedEntityForInformationRecipients()_ function.  The function will return an array of entity ids.
```
Comment (ClementBouvierN) [inline src/docs/asciidoc/reference_doc/ui_api.adoc:337]: Same as above

File: src/docs/asciidoc/reference_doc/ui_api.adoc [hunk 3]
```diff
@@ -247,10 +248,280 @@ It can be used by a template to launch some processing when loading is complete
...
+        const recipients = opfab.currentUserCard.getSelectedEntityForInformationRecipients();
+```
+
+=== Function getStartDate
+
+The template can know the current startDate of the card in creation or edition by calling the _opfab.currentUserCard.getStartDate()_ function. The function will return a number corresponding to the startDate as epoch date value.
+
+```
+        const startDate = opfab.currentUserCard.getStartDate();
+```
```
Comment (ClementBouvierN) [inline src/docs/asciidoc/reference_doc/ui_api.adoc:350]: Maybe move this section next to the getEndDate section (l. 294) ?

File: src/docs/asciidoc/reference_doc/ui_api.adoc [hunk 4]
```diff
@@ -247,10 +248,280 @@ It can be used by a template to launch some processing when loading is complete
...
+
+The template can receive the emitter entity of the card by registering a listener function. The function will be called by OperatorFabric after loading the template and every time the card emitter changes (if the user can choose from multiple entities).
+
+```
+        opfab.currentUserCard.listenToEntityUsedForSendingCard((entityId) => {// do some stuff with the entity id})
+```
+
+=== Function registerFunctionToGetSpecificCardInformation
+
+Register the template function to call to get user card specific information. This function will be call by opfab when user clicks on the "preview" button. More explanation can be found in the <<user_cards, user card chapter>>.
```
Comment (ClementBouvierN) [inline src/docs/asciidoc/reference_doc/ui_api.adoc:382]: "This function will be call by opfab when user" -> "This function will be called by opfab when the user"

**Review by quinarygio — CHANGES_REQUESTED** [review]

File: src/docs/asciidoc/reference_doc/ui_api.adoc [hunk 5]
```diff
@@ -247,10 +248,280 @@ It can be used by a template to launch some processing when loading is complete
...
+
+```
+        const expirationDate = new Date().valueOf() + 10000;
+        opfab.currentUserCard.setInitialExpirationDate(expirationDate);
+```
+
+=== Function setInitialLttd
+
+From the template it is possible to set the initial values for `lttd` by calling  _opfab.currentUserCard.setInitialLttd(lttd)_ . The lttd is a number representing an epoch date value.
+
```
Comment (ClementBouvierN) [inline src/docs/asciidoc/reference_doc/ui_api.adoc:458]: "to set the initial values" -> "to set the initial value"

File: src/docs/asciidoc/reference_doc/ui_api.adoc [hunk 6]
```diff
@@ -247,10 +248,280 @@ It can be used by a template to launch some processing when loading is complete
...
+```
+        const lttd = new Date().valueOf() + 10000;
+        opfab.currentUserCard.setInitialLttd(lttd);
+```
+
+=== Function setInitialSelectedRecipients
+
+It is possible to configure the list of initially selected recipients via javascript in the user template by
+calling the setInitialSelectedRecipients method. The method takes as
+input the list of Entity ids to be preselected.
```
Comment (ClementBouvierN) [inline src/docs/asciidoc/reference_doc/ui_api.adoc:469]: why is there line breaks in this description ?

File: src/docs/asciidoc/reference_doc/ui_api.adoc [hunk 7]
```diff
@@ -247,10 +248,280 @@ It can be used by a template to launch some processing when loading is complete
...
+        ]);
+....
+
+
+In this example the dropdown list of available recipients will have "ENTITY_FR" and "IT_SUPERVISOR_ENTITY" preselected. The user can anyway change the selected recipients.
+
+=== Function setInitialSelectedRecipientsForInformation
+
+It is possible to configure the list of initially selected recipients for information via javascript in the user template by calling the setInitialSelectedRecipientsForInformation method. The method takes as
+input the list of Entity ids to be preselected.
```
Comment (ClementBouvierN) [inline src/docs/asciidoc/reference_doc/ui_api.adoc:486]: same as above

File: src/docs/asciidoc/reference_doc/ui_api.adoc (hunk 3 above)
Comment (freddidierRTE) [reply src/docs/asciidoc/reference_doc/ui_api.adoc:350]: i will keep the alphabetical order

**Review by quinarygio — APPROVED** [review]

**Pull request merged**